### PR TITLE
[AIRFLOW-5942] Pin PyMSSQL to <3.0

### PR DESCRIPTION
**Pull request by kaxil**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. 
  - https://issues.apache.org/jira/browse/AIRFLOW-5942

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Pymssql authors have just pushed a release to PyPi that is purposefully non-functional https://github.com/pymssql/pymssql/issues/668

Unfortunately, our version constraint on it from Airflow is a bit too broad, so this will be installed if you don't do anything else when you do `pip install 'apache-airfow[mssql]'. The workaround if you need this dep is to do `pip install 'mssql~=2.1' 'apache-airfow[mssql]'`

Currently, it would error with the following:

```
Building wheels for collected packages: pymssql
  Building wheel for pymssql (PEP 517) ... error
  Complete output from command /usr/local/bin/python /usr/local/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py build_wheel /tmp/tmpp99lgy9s:
  running bdist_wheel
  running build
  installing to build/bdist.linux-x86_64/wheel
  running install
  /tmp/pip-build-env-b9denqex/overlay/lib/python3.6/site-packages/setuptools/dist.py:46: DistDeprecationWarning: Do not call this function
    warnings.warn("Do not call this function", DistDeprecationWarning)
  Traceback (most recent call last):
    File "/usr/local/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py", line 207, in <module>
      main()
    File "/usr/local/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py", line 197, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/usr/local/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py", line 141, in build_wheel
      metadata_directory)
    File "/tmp/pip-build-env-b9denqex/overlay/lib/python3.6/site-packages/setuptools/build_meta.py", line 209, in build_wheel
      wheel_directory, config_settings)
    File "/tmp/pip-build-env-b9denqex/overlay/lib/python3.6/site-packages/setuptools/build_meta.py", line 194, in _build_with_temp_dir
      self.run_setup()
    File "/tmp/pip-build-env-b9denqex/overlay/lib/python3.6/site-packages/setuptools/build_meta.py", line 237, in run_setup
      self).run_setup(setup_script=setup_script)
    File "/tmp/pip-build-env-b9denqex/overlay/lib/python3.6/site-packages/setuptools/build_meta.py", line 142, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 493, in <module>
      tests_require=['pytest', 'unittest2'],
    File "/tmp/pip-build-env-b9denqex/overlay/lib/python3.6/site-packages/setuptools/__init__.py", line 145, in setup
      return distutils.core.setup(**attrs)
    File "/usr/local/lib/python3.6/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/usr/local/lib/python3.6/distutils/dist.py", line 955, in run_commands
      self.run_command(cmd)
    File "/usr/local/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/tmp/pip-build-env-b9denqex/overlay/lib/python3.6/site-packages/wheel/bdist_wheel.py", line 228, in run
      self.run_command('install')
    File "/usr/local/lib/python3.6/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/local/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "setup.py", line 449, in run
      raise DeprecationWarning(msg)
  DeprecationWarning: The pymssql project has been discontinued.  To install the last working released version, use a
  version specifier like "pymssql<3.0".  For details and alternatives see:
  https://github.com/pymssql/pymssql/issues/668
```

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by kaxil** [comment]
Whoops wait. Didn't add Jira

File: setup.py [hunk 1]
```diff
@@ -237,7 +237,7 @@ def write_version(filename: str = os.path.join(*["airflow", "git_version"])):
 kubernetes = ['kubernetes>=3.0.0',
               'cryptography>=2.0.0']
 ldap = ['ldap3>=2.5.1']
-mssql = ['pymssql>=2.1.1']
+mssql = ['pymssql<3.0']
```
Comment (ashb) [inline setup.py]:
```suggestion
mssql = ['pymssql>=2.1.1,<3.0']
```

Comment (ashb) [reply setup.py]: Or 
```suggestion
mssql = ['pymssql~=2.1.1']
```

**Review by ashb — CHANGES_REQUESTED** [review]

**Review by ashb — APPROVED** [review]
~But lets check that this~

File: setup.py (hunk 1 above)
Comment (potiuk) [reply setup.py]: Yep.  ~= 2.1.1 should work as equivalent of 2.1.*

**Review by potiuk — APPROVED** [review]

**Pull request merged**